dev-infrastructure/scripts/kv-copy-cert.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+# Get certificate from a keyvaultS, grant required permissions
+# Grant Secret Officer role to current user, if it is not set
+# Grant Certificate Officer role to current user, if it is not set
+# Write certificate to keyvault
+# Revoke Certificates Officer role from current user, if it was added by this script
+
+if [ "$#" -ne 4 ]; then
+    echo "Usage: $0 <source-kv-name> <resource-group> <secret-name> <target-kv-value>"
+    exit 1
+fi
+
+source_kv_name=$1
+rg=$2
+secret_name=$3
+target_kv_name=$4
+
+function officer_cert_count() {
+    az role assignment list --scope ${target_kv_id} --assignee ${currentuser_client_id} --role "Key Vault Certificates Officer" | jq -r '.[].id'| wc -l
+}
+# Check and grant permissions
+permission_granted=false
+currentuser_client_id=$(az ad signed-in-user show -o json | jq -r '.id')
+source_kv_id=$(az keyvault show --name ${source_kv_name} --query id -o tsv)
+target_kv_id=$(az keyvault show --name ${target_kv_name} --resource-group ${rg} -o json | jq -r '.id')
+
+
+asign_exists=$(officer_cert_count)
+if [ $asign_exists -eq 0 ]; then
+    echo "Assigning Key Vault Certificate Officer role to current user"
+    az role assignment create --assignee ${currentuser_client_id} --role "Key Vault Certificates Officer" --scope ${target_kv_id}
+    permission_granted=true
+fi
+
+while [ $(officer_cert_count) -eq 0 ]; do
+    sleep 2
+done
+
+# Get the certificate from source kv
+az keyvault secret show --vault-name ${source_kv_name} --name ${secret_name} --query "value" -o tsv | base64 -d > /tmp/${secret_name}.pfx
+
+# Set cerrtificate on target kv
+az keyvault certificate import --vault-name ${target_kv_name} --name ${secret_name} --file /tmp/${secret_name}.pfx
+
+# Remove temporary cert
+rm /tmp/${secret_name}.pfx
+
+# Revoke secret permissions
+if [ $permission_granted == "true" ]; then
+    echo "Revoking Key Vault Secrets Officer role from current user"
+    az role assignment delete --assignee ${currentuser_client_id} --role "Key Vault Secrets Officer" --scope ${target_kv_id}
+fi
```

Comment:
Sorry, this was a mistake. It should revoke the Key Vault Certificates Officer.
The user should have the Secrets User role or the Officer one for the source keyvault.

 Is it to ensure role assignments are cleaned up to help with not reaching the max limit?
Exactly, by default you don't have that role on this scope.

Reply:
The user may not necessarily have the Secrets User/Officer role for the source kv. By default, users don't have this role assigned to them. That would be a pre-requisite of this. It's fine if it's not done here, but it would need to be documented. 

Otherwise, we should do the same as what's done for the certs officer role. Check if the role assignment is there and if not add it for the user to the source kv and remove it at the end of the script.

Reply:
Hmm, I just checked there, looks like it is there 👍 I remember having to add my role awhile back but looks like that has changed! Thanks, you can ignore this then!